pkg/operator/staticpod/controller/monitoring/monitoring_resource_controller.go

```diff
@@ -0,0 +1,204 @@
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+	"github.com/openshift/library-go/pkg/operator/staticpod/controller/common"
+	"github.com/openshift/library-go/pkg/operator/staticpod/controller/monitoring/bindata"
+)
+
+const (
+	defaultMonitoringNamespace                        = "openshift-monitoring"
+	operatorStatusMonitoringResourceControllerFailing = "MonitoringResourceControllerFailing"
+	controllerWorkQueueKey                            = "key"
+	manifestDir                                       = "pkg/operator/staticpod/controller/monitoring"
+)
+
+type MonitoringResourceController struct {
+	targetNamespace    string
+	serviceMonitorName string
+
+	saListerSynced cache.InformerSynced
+	saLister       corelisterv1.ServiceAccountLister
+
+	clusterRoleBindingLister       rbaclisterv1.ClusterRoleBindingLister
+	clusterRoleBindingListerSynced cache.InformerSynced
+
+	// queue only ever has one item, but it has nice error handling backoff/retry semantics
+	queue workqueue.RateLimitingInterface
+
+	kubeClient           kubernetes.Interface
+	dynamicClient        dynamic.Interface
+	operatorConfigClient common.OperatorClient
+	eventRecorder        events.Recorder
+}
+
+// NewMonitoringResourceController creates a new backing resource controller.
+func NewMonitoringResourceController(
+	targetNamespace string,
+	serviceMonitorName string,
+	operatorConfigClient common.OperatorClient,
+	kubeInformersForTargetNamespace informers.SharedInformerFactory,
+	kubeClient kubernetes.Interface,
+	dynamicClient dynamic.Interface,
+	eventRecorder events.Recorder,
+) *MonitoringResourceController {
+	c := &MonitoringResourceController{
+		targetNamespace:      targetNamespace,
+		operatorConfigClient: operatorConfigClient,
+		eventRecorder:        eventRecorder,
+		serviceMonitorName:   serviceMonitorName,
+
+		clusterRoleBindingListerSynced: kubeInformersForTargetNamespace.Core().V1().ServiceAccounts().Informer().HasSynced,
+		clusterRoleBindingLister:       kubeInformersForTargetNamespace.Rbac().V1().ClusterRoleBindings().Lister(),
+
+		queue:         workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "MonitoringResourceController"),
+		kubeClient:    kubeClient,
+		dynamicClient: dynamicClient,
+	}
+
+	operatorConfigClient.Informer().AddEventHandler(c.eventHandler())
+
+	// TODO: We need a dynamic informer here to observe changes to ServiceMonitor resource.
+
+	kubeInformersForTargetNamespace.Rbac().V1().ClusterRoleBindings().Informer().AddEventHandler(c.eventHandler())
+	return c
+}
+
+func (c MonitoringResourceController) mustTemplateAsset(name string) ([]byte, error) {
+	config := struct {
+		TargetNamespace     string
+		MonitoringNamespace string
+		ServiceMonitorName  string
+	}{
+		TargetNamespace:     c.targetNamespace,
+		ServiceMonitorName:  c.serviceMonitorName,
+		MonitoringNamespace: defaultMonitoringNamespace,
```

Comment:
wouldn't mind to have this hard-coded in the template. We don't support customization anyway.

Reply:
agree, if this change we can change in manifest